install.ps1

```diff
@@ -1,429 +1,888 @@
+        $numSelectedLabel.text = "Total: $($selectedPackagesBox.Items.count)"
+    }
 
-# Needed for many applications
-# Set up the correct feed
-$fireeyeFeed = "https://www.myget.org/F/fireeye/api/v2"
-iex "choco sources add -n=fireeye -s $fireeyeFeed --priority 1"
-iex "choco upgrade -y vcredist-all.flare"
-iex "refreshenv"
-
-if ($null -eq $loaded_profile) {
-  # Default install
-  Write-Host "[+] Performing normal installation..."
-  $startPath = Join-Path ${Env:ProgramData} "Microsoft\Windows\Start Menu\Programs\FLARE"
-  if (-Not (Set-EnvironmentVariableWrap "TOOL_LIST_DIR" $startPath)) {
-    Write-Warning "Failed to set environment variable TOOL_LIST_DIR"
-  }
-
-  $desktopShortcut = Join-Path ${Env:UserProfile} "Desktop\FLARE.lnk"
-  if (-Not (Set-EnvironmentVariableWrap "TOOL_LIST_SHORTCUT" $desktopShortcut)) {
-    Write-Warning "Failed to set environment variable TOOL_LIST_SHORTCUT"
-  }
-
-  choco upgrade -y -f common.fireeye
-  if (([Environment]::OSVersion).Version.Major -eq 10) {
-    choco upgrade -y flarevm.win10.preconfig.fireeye
-    if ($norestart) {
-      Install-BoxStarterPackage -PackageName "flarevm.win10.installer.fireeye" -DisableReboots
-    } else {
-      Install-BoxStarterPackage -PackageName "flarevm.win10.installer.fireeye" -Credential $cred
+    function Remove-AllPackages {
+        foreach($package in $selectedPackagesBox.Items)
+        {
+            $unselectedPackagesBox.BeginUpdate()
+            $unselectedPackagesBox.Items.Add($package) | Out-Null
+            $unselectedPackagesBox.EndUpdate()
+        }
+        $numUnselectedLabel.text = "Total: $($unselectedPackagesBox.Items.count)"
+
+        $selectedPackagesBox.BeginUpdate()
+        $selectedPackagesBox.Items.Clear()
+        $selectedPackagesBox.EndUpdate()
+        $numSelectedLabel.text = "Total: $($selectedPackagesBox.Items.count)"
     }
-  } else {
-    if ($norestart) {
-      Install-BoxStarterPackage -PackageName "flarevm.installer.flare" -DisableReboots
+
+    # Gather lists of packages (i.e., available, already installed, to install)
+    $excludedPackages = @("flarevm.installer.vm", "common.vm")
+    $installedPackages = (Get-InstalledPackages).Name
+    $packagesToInstall = $configXml.config.packages.package.name | Where-Object { $installedPackages -notcontains $_ }
+    $allPackages = (Get-InstallablePackages).Name | Where-Object { $packagesToInstall -notcontains $_ -and $installedPackages -notcontains $_ -and $excludedPackages -notcontains $_}
+    $envs = [ordered]@{}
+    $configXml.config.envs.env.ForEach({ $envs[$_.name] = $_.value })
+
+    $form                   = New-Object system.Windows.Forms.Form
+    $form.ClientSize        = New-Object System.Drawing.Point(717,740)
+    $form.text              = "FLARE VM Install Customization"
+    $form.TopMost           = $true
+    $form.MaximizeBox       = $false
+    $form.FormBorderStyle   = 'FixedDialog'
+    $form.StartPosition     = 'CenterScreen'
```

Comment:
Can the window be configured to be resizable or does that cause other layout issues?

Reply:
Since everything is a fixed coordinate, when we resize the window the buttons and fields will remain where they are relative to the top left corner -- see an example below:

![Screen Shot 2022-11-29 at 10 31 28](https://user-images.githubusercontent.com/20993114/204572176-dbf3c7f9-7d9b-4dd3-a186-52e056b02b3c.png)